packages/core/src/lib/waku_peer_exchange/peer_discovery.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  PeerDiscovery,
+  PeerDiscoveryEvents,
+  symbol,
+} from "@libp2p/interface-peer-discovery";
+import { PeerInfo } from "@libp2p/interface-peer-info";
+import { EventEmitter } from "@libp2p/interfaces/dist/src/events";
+import debug from "debug";
+
+import { multiaddrsToPeerInfo } from "../multiaddr_to_peer_info";
+
+import { WakuPeerExchange } from ".";
+
+const log = debug("waku:peer-discovery-dns");
+
+export class PeerExchangeDiscovery
+  extends EventEmitter<PeerDiscoveryEvents>
+  implements PeerDiscovery
+{
+  private _started: boolean;
+
+  constructor(public peerExchange: WakuPeerExchange, public numPeers: number) {
+    super();
+    this._started = false;
+  }
+
+  /**
+   * Start discovery process
+   */
+  async start(): Promise<void> {
+    log("Starting peer discovery via Peer Exchange");
+
+    this._started = true;
+
+    const response = await this.peerExchange.query({
+      numPeers: BigInt(this.numPeers),
+    });
+
+    response.peerInfos.forEach((peerInfo) => {
+      if (!peerInfo.ENR) return;
+
+      const _peerInfos = multiaddrsToPeerInfo(peerInfo.ENR.getFullMultiaddrs());
+
+      _peerInfos.forEach((peerInfo) => {
+        if (!this._started) return;
+        this.dispatchEvent(
+          new CustomEvent<PeerInfo>("peer", { detail: peerInfo })
+        );
```

Comment:
This won't work but when you start the service, it will do one request and stop.

Instead, it should do a "regular" querying for peers.

What I would suggest is:

1. Do a peer query every time we connect to a new peer with peer exchange capability
2. Then, for the given peer, do a follow up query every X seconds, the interval to be set via `options` on constructor.

Reply:
addressed in this PR. lmk if looks good